### PR TITLE
Serialize NDArray name

### PR DESCRIPTION
**Pull request by frankfliu**
## Channel for questions ##
Before or while filing an issue please feel free to join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w) to get in touch with development team, ask questions, find out what's cooking and more!

## Description ##
(Brief description of what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- [ ] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
    - For new examples, README.md is added to explain the what the example does.
- [ ] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, Java doc)
- [ ] Feature2, tests, (and when applicable, Java doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: integration/src/main/java/ai/djl/integration/tests/ndarray/NDArrayOtherOpTest.java [hunk 1]
```diff
@@ -727,10 +728,22 @@ public void testArgMin() {
     @Test
     public void testEncodeDecode() {
         try (NDManager manager = NDManager.newBaseManager()) {
-            NDArray array = manager.create(new long[] {0, 3, 4, 2}, new Shape(2, 2));
+            NDArray array = manager.create(new byte[] {0, 3}, new Shape(2));
```
Comment (stu1130) [inline integration/src/main/java/ai/djl/integration/tests/ndarray/NDArrayOtherOpTest.java:731]: is there any reason to change the dtype to int8 here?

Comment (frankfliu) [reply integration/src/main/java/ai/djl/integration/tests/ndarray/NDArrayOtherOpTest.java:731]: No, just trying to reduce the size of encoded buffer in the test.

**Review by stu1130 — APPROVED** [review]
